tutor/src/components/exercises/cards.jsx

```diff
@@ -30,10 +30,18 @@ class SectionsExercises extends React.Component {
     onExerciseToggle:           PropTypes.func,
   };
 
+  @computed get exercises() {
+    return exerciseSort(
+      this.props.exercises.byPageId[this.props.pageId],
+      this.props.course,
+      User,
+    );
+  }
+
   render() {
-    const { pageId, book, exercises, ...previewProps } = this.props;
-    const page = book.pages.byId.get(pageId);
-    const sectionExercises = exercises.byPageId[pageId];
+    const { pageId, course, ...previewProps } = this.props;
+    const sectionExercises = this.exercises;
+    const page = course.referenceBook.pages.byId.get(pageId);
```

Comment:
I ran into issues sometimes where `course.referenceBook` is `undefined` because it has not started a fetch yet for the book. Should we ensure it is fetched before it gets into this part of the component?

Reply:
yes good idea, I'll add a guard for that

Reply:
after looking at it, referenceBOok will always be set.  when it's not loaded page won't be found though, so I'll check for that